frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -161,6 +162,19 @@ def ensure_sufficient_stock_lines
       end
     end
 
+    def ensure_sufficient_shipment_stock_quantities
+      if @order.passed_checkout_step?('delivery')
+        availability_validator = Spree::Stock::AvailabilityValidator.new
+        unless @order.line_items.all? { |item| availability_validator.validate(item) }
+          unavailable_items = @order.line_items.map { |item| item.errors.full_messages.to_sentence }.to_sentence
+          flash[:error] = t('spree.inventory_error_flash_for_insufficient_shipment_quantity', unavailable_items: unavailable_items)
+          @order.restart_checkout_flow
+          @order.next!
+          redirect_to spree.checkout_state_path(:delivery)
```

Comment:
Yes, the goal is to revert the order status to `delivery`, and there is no apparent need to restart from the `cart` state, but my guess is that using `#restart_checkout_flow` and add one `#next!` call may be better, at least in the long run, as `#ensure_updated_shipments` might develop in a more complex method that adds more (required) behavior to the restarting process. I'd rather follow the path than create a new one with code such as `@order.update state: :delivery` or someting similar.

On the other side what makes me frown a bit is the `@order.next!` line, as it may have one meaning today (advance the order state to `delivery`) but that can change in the future. 

Maybe the best thing would be to `restart_checkout_flow` and remove the `next!`.

I'm not sure about what's the best solution, so I'm committed to following the advice of people with a better understanding of the checkout process and its intricacies.